autoload/sj/yaml.vim

```diff
@@ -139,5 +139,87 @@ function! s:GetChildren(line_no)
   endwhile
   let next_line_no = next_line_no - 1
 
+  " Preserve trailing empty lines
+  while sj#BlankString(getline(next_line_no)) && next_line_no > line_no
+    let next_line_no = next_line_no - 1
+  endwhile
+
   return [sj#GetLines(line_no + 1, next_line_no), next_line_no]
 endfunction
+
+function! s:splitArrayLines(array)
+  let lines = []
+
+  let line = ''
+  let fences = { '"': '"', "'": "'", '{': '}' }
+
+  for chunk in split(a:array, ',')
+    if line != ''
+      " Inside a string
+      let line = line . ',' . chunk
+
+      if chunk =~ fences[line[0]] . '\s*$'
+        " End of string
+        call add(lines, s:convertToSingleLine(line))
+        let line  = ''
+      endif
+
+    " Start of string
+    elseif chunk =~ '^\s*[' . join(keys(fences), '') . ']'
+      let line  = sj#Ltrim(chunk)
+
+      if chunk =~ fences[line[0]] . '\s*$'
+        " Complete string
+        call add(lines, s:convertToSingleLine(line))
+        let line  = ''
+      endif
+    else
+      call add(lines, s:convertToSingleLine(chunk))
+    endif
+  endfor
+
+  if line != ''
+    cal add(lines, s:convertToSingleLine(line))
+  endif
+
+  return lines
+endfunction
+
+function! s:joinArrayLines(lines)
+  let lines = map(a:lines, 's:convertToCompactLine(v:val)')
+  return join(lines, ', ')
+endfunction
+
+function! s:convertToCompactLine(line)
+  let prop = substitute(a:line, '\v^([a-z]+):.*', '\1', 1)
+  if prop != a:line
+    return '{ ' . a:line . ' }'
+  endif
+
+  return a:line
+endfunction
+
+function! s:convertToSingleLine(line)
+  let line = sj#Trim(a:line)
+
+  if line =~ '^{.*}$'
+    let parser = sj#argparser#js#Construct(2, len(line) - 1, line)
+    call parser.Process()
+
+    if len(parser.args) == 1
+      let line = substitute(line, '^{\s*', '', '')
+      let line = substitute(line, '\s*}$', '', '')
+    endif
+  endif
+
+  return line
+endfunction
```

Comment:
I feel like these function names are a bit iffy. If I understand correctly, `convertToCompactLine` checks if it should add `{}` around a key-value pair, while `convertToSingleLine` does the opposite. Maybe better names would be `addCurlyBrackets` and `removeCurlyBrackets`? Or `wrapCurlyBrackets`/`stripCurlyBrackets`? Could also be `tryWrap`/`tryAdd`/`maybeWrap`/`maybeAdd`. Basically, it seems like the add of packaging or removing brackets is the goal of the function, and while "convertSingle" or "convertCompact" is a more generic term, it's also maybe too generic -- it took me some time to figure out what the functions were doing, exactly. (Also a bit confusing to me, since a single line should be the more compact representation, but in this case, "compact" would be for the other way around?)

Reply:
You are right, all of you suggestions are better than what I chose :-D